staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/mutating/metrics/metrics.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	metricsNamespace = "apiserver"
+	metricsSubsystem = "mutating_admission_policy"
+)
+
+// MutationErrorType defines different error types that happen to a mutation expression
+type MutationErrorType string
+
+const (
+	// MutationCompileError indicates that the expression fails to compile.
+	MutationCompileError MutationErrorType = "compile_error"
+	// MutatingInvalidError indicates that the expression fails due to internal
+	// errors that are out of the control of the user.
+	MutatingInvalidError MutationErrorType = "invalid_error"
+	// MutatingOutOfBudget indicates that the expression fails due to running
+	// out of cost budget, or the budget cannot be obtained.
+	MutatingOutOfBudget MutationErrorType = "out_of_budget"
+	// MutationNoError indicates that the expression returns without an error.
+	MutationNoError MutationErrorType = "no_error"
+)
+
+var (
+	// Metrics provides access to mutation admission metrics.
+	Metrics = newMutationAdmissionMetrics()
+)
+
+// MutatingAdmissionPolicyMetrics aggregates Prometheus metrics related to mutation admission control.
+type MutatingAdmissionPolicyMetrics struct {
+	policyCheck   *metrics.CounterVec
+	policyLatency *metrics.HistogramVec
+}
+
+func newMutationAdmissionMetrics() *MutatingAdmissionPolicyMetrics {
+	check := metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Namespace:      metricsNamespace,
+			Subsystem:      metricsSubsystem,
+			Name:           "check_total",
+			Help:           "Mutation admission policy check total, labeled by policy and further identified by binding.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"policy", "policy_binding", "mutation_index", "error_type"},
```

Comment:
is mutation_index a helpful label here? This can potentially be unbounded, is that right?

Reply:
I just added it cause I feel user might want to know which mutation showed in the metrics. It should not unbounded due to the size limit. I can remove it if that's a concern^^

Reply:
Removed^